File: C#/1 - Single-Principle/Caso Certo/Curso.cs
namespace PlataformaDeEnsino.Certo;

using PlataformaDeEnsino.Common;

public class Curso
{
    public int Id { get; set; }
    public string Nome { get; set; } = "";
    public Professor Professor { get; set; } = new Professor();

    public bool Validar(out string errorMessage)
    {
        errorMessage = "";

        if (String.IsNullOrEmpty(Nome))
            errorMessage = "Nome não pode ser vazio";

        return errorMessage.Length > 0;
    }
}

File: Java/PlataformaDeEnsino/Certo/SinglePrinciple/Curso.java
package PlataformaDeEnsino.Certo.SinglePrinciple;

import PlataformaDeEnsino.Certo.LiskovPrinciple.Professor;

public class Curso {
    private int id;
    private String nome;
    private Professor professor;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Professor getProfessor() {
        return professor;
    }

    public void setProfessor(Professor professor) {
        this.professor = professor;
    }

    public String getErrors() {
        if (nome == null || nome.trim().isEmpty())
            return "O nome do curso não pode ser nulo";
        
        return null;
    }

    public Boolean isValido() {
        if (nome == null || nome.trim().isEmpty())
            return false;

        return true;
    }
}
